props/blaster.h

```diff
@@ -200,76 +200,6 @@ class Blaster : public PROP_INHERIT_PREFIX PropBase {
     }
   }
 
-  // Self-destruct pulled from detonator 
-  bool armed_ = false;
-
-  enum NextAction {
-    NEXT_ACTION_NOTHING,
-    NEXT_ACTION_ARM,
-    NEXT_ACTION_BLOW,
-  };
-
-  NextAction next_action_ = NEXT_ACTION_NOTHING;
-  uint32_t time_base_;
-  uint32_t next_event_time_;
-  
-  void SetNextAction(NextAction what, uint32_t when) {
-    time_base_ = millis();
-    next_event_time_ = when;
-    next_action_ = what;
-  }
-
-  void SetNextActionF(NextAction what, float when) {
-    SetNextAction(what, when * 1000);
-  }
-  
-  void PollNextAction() {
-    if (millis() - time_base_ > next_event_time_) {
-      switch (next_action_) {
-        case NEXT_ACTION_NOTHING:
-          break;
-        case NEXT_ACTION_ARM:
-          armed_ = true;
-          // TODO: Should we have separate ARMING and ARMED states?
-          break;
-        case NEXT_ACTION_BLOW:
-          Off(OFF_BLAST);
-          break;
-      }
-      next_action_ = NEXT_ACTION_NOTHING;
-    }
-  }
-  
-  void beginArm() {
-    SaberBase::SetLockup(SaberBase::LOCKUP_ARMED);
-    SaberBase::DoBeginLockup();
-#ifdef ENABLE_AUDIO    
-    float len = hybrid_font.GetCurrentEffectLength();
-#else    
-    float len = 1.6;
-#endif
-    SetNextActionF(NEXT_ACTION_ARM, len);
-  }
-
-  void selfDestruct() {
-    SaberBase::DoEndLockup();
-#ifdef ENABLE_AUDIO    
-    float len = hybrid_font.GetCurrentEffectLength();
-#else    
-    float len = 0.0;
-#endif
-    SaberBase::SetLockup(SaberBase::LOCKUP_NONE);
-    if (armed_) {
-      SetNextActionF(NEXT_ACTION_BLOW, len);
-    } else {
-      SetNextAction(NEXT_ACTION_NOTHING, 0);
-    }
-  }
-
-  void Loop() override {
```

Comment:
It might be better to only remove the Loop() function.

That way, a prop that inherits from blaster.h could just add PollNextAction() to their loop function, and call SelfDestruct() from Event2() to implement selfdestruct.

Code that is never called anywhere is optimized out, so the rest of this code would just get optimized away if nothing calls it.

We have similar things going on in saber.h, there are lots of functions in there which are never called from saber.h, but called from lots of subclasses.